### PR TITLE
Add pulp_deb for pulpcore

### PR DESCRIPTION
**Pull request by m-bucher**
Adding latest beta-version of pulp_deb for pulpcore (pulp3)

Request for Comments (e.g. from @quba42 and @mdellweg)
Is python-debian the only dependency we need?

### CODE REVIEW

**Comment by quba42** [comment]
As far as I know, `python-debian` is the only non-pulp dependency.

**Comment by mdellweg** [comment]
Yes i think that is right. Sounds pretty neat when you formulate it that way.
There will be one more dependency on the postgresql function, once that work is included.

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+License:        GPLv2+
+URL:            https://pulpproject.org/#deb
+Source0:        https://files.pythonhosted.org/packages/source/p/%{pypi_name}/%{pypi_name}-%{version}.tar.gz
+BuildArch:      noarch
+
+BuildRequires:  python3-devel
+BuildRequires:  python3-pulpcore < 3.4
+BuildRequires:  python3-pulpcore >= 3.3
+BuildRequires:  python3-debian >= 0.1.36
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:17]: if you drop these buildrequires, the package will build fine without spliting the PR into two ;)

File: comps/comps-katello-pulpcore-el7.xml [hunk 2]
```diff
@@ -24,6 +24,7 @@
       <packagereq type="default">python3-coreapi</packagereq>
       <packagereq type="default">python3-coreschema</packagereq>
       <packagereq type="default">python3-cryptography</packagereq>
+      <packagereq type="default">python3-debian</packagereq>
```
Comment (evgeni) [inline comps/comps-katello-pulpcore-el7.xml:27]: can you add the same to the EL8 comps?

**Comment by evgeni** [comment]
blocked on  #5127 for new pulpcore

**Comment by evgeni** [comment]
cc @jlsherrill as you asked about deb yesterday :)

File: packages/pulpcore/python-debian/python-debian.spec [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+rm -rf %{pypi_name}.egg-info
+
+%build
+%py3_build
+
+%install
+%py3_install
+
+%check
+%{__python3} setup.py test
```
Comment (evgeni) [inline packages/pulpcore/python-debian/python-debian.spec:55]: the tests seem to fail and we've been skipping tests for the other pulpcore packages too, so feel free to drop this

Comment (m-bucher) [inline packages/pulpcore/python-debian/python-debian.spec:55]: Apparently `python-debian` does not deliver the test-files in the package.
So I guess I will remove this!?

Comment (evgeni) [reply packages/pulpcore/python-debian/python-debian.spec:55]: Yeah, just nuke it :)

File: comps/comps-katello-pulpcore-el7.xml (hunk 2 above)
Comment (m-bucher) [reply comps/comps-katello-pulpcore-el7.xml:27]: Yes, done :-)

File: packages/pulpcore/python-debian/python-debian.spec (hunk 3 above)
Comment (m-bucher) [reply packages/pulpcore/python-debian/python-debian.spec]: with pleasure :-)

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec (hunk 1 above)
Comment (m-bucher) [reply packages/pulpcore/python-pulp-deb/python-pulp-deb.spec]: done

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+%build
+%py3_build
+
+%install
+%py3_install
+
+%check
+%{__python3} setup.py test
+
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:45]:
```suggestion
```

same, drop the tests ;)

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Created by pyp2rpm-3.3.2
+%global pypi_name pulp-deb
+
+Name:           python-%{pypi_name}
+Version:        2.3.0b1
+Release:        1%{?dist}
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:6]:
```suggestion
Version:        2.3.0
Release:        0.1.b1%{?dist}
```

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Summary:        pulp-deb plugin for the Pulp Project
+
+License:        GPLv2+
+URL:            https://pulpproject.org/#deb
+Source0:        https://files.pythonhosted.org/packages/source/p/%{pypi_name}/%{pypi_name}-%{version}.tar.gz
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:11]:
```suggestion
Source0:        https://files.pythonhosted.org/packages/source/p/%{pypi_name}/%{pypi_name}-%{version}b1.tar.gz
```

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+Requires:       python3-pulpcore < 3.4
+Requires:       python3-pulpcore >= 3.3
+Requires:       python3-debian >= 0.1.36
+Requires:       python3-setuptools
+%description -n python3-%{pypi_name}
+
+
+
+%prep
+%autosetup -n %{pypi_name}-%{version}
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:33]:
```suggestion
%autosetup -n %{pypi_name}-%{version}b1
```

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+%build
+%py3_build
+
+%install
+%py3_install
+
+%files -n python3-%{pypi_name}
+%doc README.rst
+%{python3_sitelib}/pulp_deb
+%{python3_sitelib}/pulp_deb-%{version}-py?.?.egg-info
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:46]:
```suggestion
%{python3_sitelib}/pulp_deb-%{version}b1-py?.?.egg-info
```

File: packages/pulpcore/python-pulp-deb/python-pulp-deb.spec [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+%changelog
+* Thu Apr 30 2020 ATIX AG <info@atix.de> - 2.3.0b1-1
```
Comment (evgeni) [inline packages/pulpcore/python-pulp-deb/python-pulp-deb.spec:49]:
```suggestion
* Thu Apr 30 2020 ATIX AG <info@atix.de> - 2.3.0-0.1.b1
```

**Comment by jlsherrill** [comment]
This looks good, should it be moved out of draft?

**Comment by evgeni** [comment]
Yeah, looks good to me too.

**Comment by m-bucher** [comment]
@evgeni should we use the `prerelease`-flag here as well?
https://github.com/theforeman/foreman-packaging/pull/5165#discussion_r424121265

**Comment by evgeni** [comment]
@m-bucher the ruby rpm tools use that macro internally, the python ones do not. So yes, you could define the macro, but you'd need to do the replacements everywhere yourself. I'm fine with either.

**Comment by m-bucher** [comment]
OK, let's keep it like it is for now :smile:

**Comment by jlsherrill** [comment]
thanks @m-bucher !

**Pull request merged**